user-access-commons/src/main/java/no/unit/nva/model/UserDto.java

```diff
@@ -204,7 +204,7 @@ public int hashCode() {
     /*This exception should not happen as a RoleDb should always convert to a RoleDto */
     private static <T> IllegalStateException unexpectedException(Failure<T> failure) {
         logger.error(ERROR_DUE_TO_INVALID_ROLE);
-        throw new IllegalStateException(failure.getException());
+        return new IllegalStateException(failure.getException());
```

Comment:
bugfix. this method should actually return an exception that is thrown by the caller.